sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -52,5 +64,90 @@ public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
             Policies ??= new();
             Policies.Add((position, policy));
         }
+
+        /// <summary>
+        /// Adds a custom classifier to the <see cref="ResponseClassifier"/> that decides if the response
+        /// received from the service should be considered an error response, for this service call.
+        /// The custom classifier is applied before the default classifier.
+        /// This is useful in cases where you'd like to prevent specific response status codes from appearing as errors in
+        /// logging and distributed tracing.  It will also prevent the call from throwing an exception when a response with
+        /// this status code is received.
+        /// </summary>
+        /// <param name="statusCode">The status code to customize classification for.</param>
+        /// <param name="isError">Whether the passed-in status code should be classified as an error.</param>
+        public void AddClassifier(int statusCode, bool isError)
+        {
+            int length = _classifiers == null ? 0 : _classifiers.Length;
+            Array.Resize(ref _classifiers, length + 1);
+            _classifiers[length] = new StatusCodeClassifier(statusCode, isError);
+        }
+
+        /// <summary>
+        /// Adds a custom <see cref="MessageClassifier"/> to the <see cref="ResponseClassifier"/> that decides if the response
+        /// received from the service should be considered an error response, for this service call.
+        /// The custom classifier is applied before the default classifier.
+        /// This is useful in cases where you'd like to prevent specific response status codes from appearing as errors in
+        /// logging and distributed tracing.  It will also prevent the call from throwing an exception when a response with
+        /// this status code is received.
+        /// </summary>
+        /// <param name="classifier">The custom classifier.</param>
+        public void AddClassifier(MessageClassifier classifier)
+        {
+            int length = _classifiers == null ? 0 : _classifiers.Length;
+            Array.Resize(ref _classifiers, length + 1);
+            _classifiers[length] = classifier;
+        }
+
+        private class StatusCodeClassifier : MessageClassifier
```

Comment:
Could status code classifier store many int,bool pairs such that we don't need separate classifier instances if the use maps more than one pair? Should we use the 10 ulongs table for that?

Reply:
I wasn't able to see how the 10 ulongs table can implement TryClassify, (I can see it with 20 ulongs).  Was there something obvious I missed there?

Reply:
The 10 ulong table would at the end of the chain.

Reply:
A MessageClassifier can't be at the end of the chain, though, because it only has TryClassify semantics.  But also, I don't believe we can set the default classifier from RequestContext, because it would override any per-client default provided by the library author, or any per-operation default provided by the DPG generator from the REST API spec.

I've updated the PR description to discuss how I thought about precedence rules here: https://github.com/Azure/azure-sdk-for-net/pull/26547#issue-1111178434  If we disagree with any of those constraints, it would be good to know, and may have implications for the overall design, although I think these satisfy all scenarios we're interested in solving here.